flink-streaming-java/src/main/java/org/apache/flink/streaming/api/operators/async/queue/StreamElementQueue.java

```diff
@@ -19,68 +19,56 @@
 package org.apache.flink.streaming.api.operators.async.queue;
 
 import org.apache.flink.annotation.Internal;
+import org.apache.flink.streaming.api.functions.async.ResultFuture;
+import org.apache.flink.streaming.api.operators.TimestampedCollector;
 import org.apache.flink.streaming.api.operators.async.AsyncWaitOperator;
+import org.apache.flink.streaming.runtime.streamrecord.StreamElement;
 
-import java.util.Collection;
+import java.util.List;
+import java.util.Optional;
 
 /**
- * Interface for blocking stream element queues for the {@link AsyncWaitOperator}.
+ * Interface for a non-blocking stream element queues for the {@link AsyncWaitOperator}.
  */
 @Internal
-public interface StreamElementQueue {
+public interface StreamElementQueue<OUT> {
 
 	/**
-	 * Put the given element in the queue if capacity is left. If not, then block until this is
-	 * the case.
+	 * Try to put the given element in the queue. This operation succeeds if the queue has capacity left and fails if
+	 * the queue is full.
 	 *
-	 * @param streamElementQueueEntry to be put into the queue
-	 * @param <T> Type of the entries future value
-	 * @throws InterruptedException if the calling thread has been interrupted while waiting to
-	 * 	insert the given element
-	 */
-	<T> void put(StreamElementQueueEntry<T> streamElementQueueEntry) throws InterruptedException;
-
-	/**
-	 * Try to put the given element in the queue. This operation succeeds if the queue has capacity
-	 * left and fails if the queue is full.
+	 * <p>This method returns a handle to the inserted element that allows to set the result of the computation.</p>
```

Comment:
I guess we do not need the tail </p> before in code formatting rule. Anything changed now?

Reply:
No, IDE just automatically inserts it.